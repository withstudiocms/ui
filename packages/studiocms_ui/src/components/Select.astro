---
import Icon from '../utils/Icon.astro';
import { generateID } from '../utils/generateID';

/**
 * The props for the select component.
 */
interface Option {
    /**
     * The label of the option.
     */
	label: string;
    /**
     * The value of the option.
     */
	value: string;
    /**
     * Whether the option is disabled. Defaults to `false`.
     */
	disabled?: boolean;
}

/**
 * The props for the select component.
 */
interface Props {
    /**
     * The label of the select.
     */
	label?: string;
    /**
     * The default value of the select.
     */
	defaultValue?: string;
    /**
     * Additional classes to apply to the select.
     */
	class?: string;
    /**
     * The name of the select. Required because of the helper.
     */
	name?: string;
    /**
     * Whether the select is required. Defaults to `false`.
     */
	isRequired?: boolean;
    /**
     * The options to display in the select.
     */
	options: Option[];
    /**
     * Whether the select is disabled. Defaults to `false`.
     */
	disabled?: boolean;
    /**
     * Whether the select is full width. Defaults to `false`.
     */
	fullWidth?: boolean;
    /**
     * The placeholder of the select. Defaults to `Select`.
     */
	placeholder?: string;
}

const {
	label,
	defaultValue,
	class: className,
	name = generateID('select'),
	isRequired,
	options = [],
	disabled,
	fullWidth,
	placeholder = "Select",
} = Astro.props;
---

<div
    id={`${name}-container`}
    class="sui-select-label"
    class:list={[disabled && "disabled", className, fullWidth && "full"]}
    data-options={JSON.stringify(options)}
    data-id={name}
>
    {label && (
        <label class="label" for={`${name}-select-btn`}>
            {label}
            <span class="req-star">{isRequired && "*"}</span>
        </label>
    )}
    <button
        class="sui-select-button"
        role="combobox"
        aria-controls={`${name}-dropdown`}
        aria-expanded="false"
        id={`${name}-select-btn`}
        type="button"
        aria-label={placeholder}
        title={placeholder}
    >
        <span class="sui-select-value-span" id={`${name}-value-span`}>
            {
                defaultValue
                    ? options.find((x) => x.value === defaultValue)?.label
                    : placeholder
            }
        </span>
        <Icon name="chevron-up-down" width={24} height={24} />
    </button>
    <ul class="sui-select-dropdown" role="listbox" id={`${name}-dropdown`}>
        {
            options.map((x, i) => (
                <li
                    class="sui-select-option"
                    role="option"
                    value={x.value}
                    class:list={[
                        defaultValue === x.value && `selected`,
                        x.disabled && "disabled",
                    ]}
                    id={defaultValue === x.value ? `${name}-selected` : ""}
                    data-option-index={i}
                >
                    {x.label}
                </li>
            ))
        }
    </ul>
    <select class="sui-hidden-select" id={name} name={name} required={isRequired} hidden tabindex="-1">
        <option value={""}> Select </option>
        {
            options.map((x) => (
                <option
                    value={x.value}
                    selected={defaultValue === x.value}
                    disabled={x.disabled}
                >
                    {x.label}
                </option>
            ))
        }
    </select>
</div>
<script>
    const allSelects = document.querySelectorAll<HTMLDivElement>(".sui-select-label");
    // id, options

    for (const container of allSelects) {
        const hiddenSelect = container.querySelector('select')!;
        const button = container.querySelector('button')!;
        const valueSpan = container.querySelector('.sui-select-value-span')!;
        const dropdown = container.querySelector('.sui-select-dropdown')!;
        const optionElements = container.querySelectorAll<HTMLLIElement>('.sui-select-option');

        const options = JSON.parse(container.dataset.options!);
        const id = container.dataset.id!;
        let active = false;

        const closeDropdown = () => {
            dropdown.classList.remove("active", "above");
            active = false;
            button.ariaExpanded = 'false';
            focusIndex = -1;

            for (const entry of optionElements) {
                entry.classList.remove('focused');
            }
        };

        const openDropdown = (toggle: boolean) => {
            const { bottom, left, right, width, x, y, height } =
                button.getBoundingClientRect();

            const optionHeight = 36;
            const totalBorderSize = 2;
            const margin = 4;

            const dropdownHeight =
                options.length * optionHeight + totalBorderSize + margin;

            const CustomRect = {
                top: bottom + margin,
                left,
                right,
                bottom: bottom + margin + dropdownHeight,
                width,
                height: dropdownHeight,
                x,
                y: y + height + margin,
            };

            if (active && toggle) {
                closeDropdown();
                return;
            }

            active = true;
            button.ariaExpanded = 'true';

            // Set focusIndex to currently selected option
            focusIndex = Array.from(optionElements).findIndex((x) =>
                x.classList.contains("selected")
            );

            if (
                CustomRect.top >= 0 &&
                CustomRect.left >= 0 &&
                CustomRect.bottom <=
                    (window.innerHeight || document.documentElement.clientHeight) &&
                CustomRect.right <=
                    (window.innerWidth || document.documentElement.clientWidth)
            ) {
                dropdown.classList.add("active");
            } else {
                dropdown.classList.add("active", "above");
            }
        }

        button.addEventListener("click", () => openDropdown(true));

        let focusIndex = -1;

        const recomputeOptions = () => {
            for (const entry of optionElements) {
                if (Number.parseInt(entry.dataset.optionIndex!) == focusIndex) {
                    entry.classList.add('focused');
                } else {
                    entry.classList.remove('focused');
                }
            }
        }

        button.addEventListener('keydown', (e) => {
            if (e.key === 'Tab' || e.key === 'Escape') {
                closeDropdown();
                return;
            }

            if (e.key === ' ' && !active) openDropdown(false);

            if (e.key === 'Enter') {
                let currentlyFocused = container.querySelector<HTMLElement>('.focused');
                if (currentlyFocused) {
                    currentlyFocused.classList.remove('focused');
                    currentlyFocused.click();

                    // Stop dropdown from immediately reopening
                    e.preventDefault();
                    e.stopImmediatePropagation();
                }

                return;
            }

            e.preventDefault();
            e.stopImmediatePropagation();

            const neighbor = (offset: number) => {
                return optionElements.item((Array.from(optionElements).findIndex((x) => x.classList.contains("selected")) ?? -1) + offset)
            }

            if (e.key === "ArrowUp" && (focusIndex > 0 || !active)) {
                if (!active) return neighbor(-1)?.click();
                focusIndex--;
                recomputeOptions();
            }

            if (e.key === "ArrowDown" && focusIndex + 1 < optionElements.length) {
                if (!active) return neighbor(1)?.click();
                focusIndex++;
                recomputeOptions();
            }

            if (e.key === 'PageUp') {
                focusIndex = 0;
                if (!active) return optionElements.item(focusIndex)?.click();
                recomputeOptions();
            }
            if (e.key === 'PageDown') {
                focusIndex = optionElements.length - 1;
                if (!active) return optionElements.item(focusIndex)?.click();
                recomputeOptions();
            }
        });

        const handleSelection = (e: MouseEvent, option: HTMLElement) => {
            e.stopImmediatePropagation();
            if (option.id === `${id}-selected` || !id) return;
            const currentlySelected = document.getElementById(`${id}-selected`);
            if (currentlySelected) {
                currentlySelected.classList.remove("selected");
                currentlySelected.id = "";
            }
            option.id = `${id}-selected`;
            option.classList.add("selected");
            const opt = options[parseInt(option.dataset.optionIndex!)];
            hiddenSelect.value = opt.value;
            valueSpan.textContent = opt.label;
            closeDropdown();
        }

        optionElements.forEach((option) => {
            const handleSelectionForOption = (e: MouseEvent) => handleSelection(e, option)

            option.addEventListener("click", handleSelectionForOption);
        });

        window.addEventListener("scroll", closeDropdown);

        document.addEventListener("keydown", (e) => {
            if (e.key === "Escape" && dropdown.classList.contains("active")) {
                closeDropdown();
            }
        });

        hideOnClickOutside(container);

        function hideOnClickOutside(element: HTMLElement) {
            const outsideClickListener = (event: MouseEvent) => {
                if (
                    !element.contains(event.target as HTMLElement) &&
                    isVisible(element) &&
                    active === true
                ) {
                    // or use: event.target.closest(selector) === null
                    closeDropdown();
                }
            };

            document.addEventListener("click", outsideClickListener);
        }

        // source (2018-03-11): https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js
        const isVisible = (elem: HTMLElement) =>
            !!elem &&
            !!(
                elem.offsetWidth ||
                elem.offsetHeight ||
                elem.getClientRects().length
            );
    }
</script>
<style>
    .sui-select-label {
        width: fit-content;
        display: flex;
        flex-direction: column;
        gap: 0.25rem;
        min-width: 200px;
        position: relative;
        height: fit-content;
    }

    .sui-select-label.full, .sui-select-label.full .sui-select-button {
        width: 100%;
    }

    .sui-select-label.disabled {
        opacity: 0.5;
        pointer-events: none;
        color: hsl(var(--text-muted));
    }

    .label {
        font-size: 14px;
    }

    .req-star {
        color: hsl(var(--danger-base));
        font-weight: 700;
    }

    .sui-select-button {
        padding: 0.5rem 0.75rem 0.5rem 1rem;
        border-radius: 8px;
        border: 1px solid hsl(var(--border));
        background: hsl(var(--background-step-2));
        color: hsl(var(--text-normal));
        transition: background border 0.15s ease;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-between;
        cursor: pointer;
        gap: 1rem;
    }

    .sui-select-button:focus {
        border: 1px solid hsl(var(--primary-base));
    }       

    .sui-select-button:hover, .sui-select-button:focus {
        background: hsl(var(--background-step-3));
    }

    .sui-select-button.active,
    .sui-select-button:active,
    .sui-select-button:has(+ .sui-select-dropdown.active) {
        border: 1px solid hsl(var(--primary-base));
        background: hsl(var(--background-step-2));
    }

    .sui-select-dropdown {
        position: absolute;
        width: 100%;
        border: 1px solid hsl(var(--border));
        list-style: none;
        margin: 0;
        padding: 0;
        flex-direction: column;
        border-radius: 0.5rem;
        background-color: hsl(var(--background-step-2));
        overflow: hidden;
        top: calc(100% + 0.25rem);
        left: 0;
        display: none;
        z-index: 90;
        box-shadow: 0px 4px 8px hsl(var(--shadow), 0.5);
    }

    .sui-select-dropdown.active {
        display: flex;
    }

    .sui-select-dropdown.above {
        top: auto;
        bottom: calc(100% - 18px + 0.25rem);
    }

    .sui-select-option {
        padding: 0.5rem;
        cursor: pointer;
        font-size: 0.975em;
        transition: all 0.15s ease;
    }

    .sui-select-option.disabled {
        pointer-events: none;
        color: hsl(var(--text-muted));
    }

    .sui-select-option:hover, .sui-select-option:focus, .sui-select-option.focused {
        background-color: hsl(var(--background-step-3));
    }

    .sui-select-option:focus {
        outline: none;
        border: none;
    }

    .sui-select-option.selected {
        background-color: hsl(var(--primary-base));
        color: hsl(var(--text-inverted));
        cursor: default;
    }

    .sui-hidden-select {
        height: 0;
        width: 0;
        border: none;
        outline: none;
        position: absolute;
        background-color: transparent;
        pointer-events: none;
        opacity: 0;
    }
</style>
