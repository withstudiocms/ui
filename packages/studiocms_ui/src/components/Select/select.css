.sui-select-label {
  width: fit-content;
  display: flex;
  flex-direction: column;
  gap: 0.25rem;
  min-width: 200px;
  position: relative;
  height: fit-content;
}

.sui-select-label.full, .sui-select-label.full .sui-select-button {
  width: 100%;
}

.sui-select-label.disabled {
  opacity: 0.5;
  pointer-events: none;
  color: hsl(var(--text-muted));
}

.label {
  font-size: 14px;
}

.req-star {
  color: hsl(var(--danger-base));
  font-weight: 700;
}

.sui-select-button {
  padding: 0.5rem 0.75rem 0.5rem 1rem;
  border-radius: var(--radius-md);
  border: 1px solid hsl(var(--border));
  background: hsl(var(--background-step-2));
  color: hsl(var(--text-normal));
  transition: background border 0.15s ease;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  gap: 1rem;
}

.sui-select-button:focus {
  border: 1px solid hsl(var(--primary-base));
}       

.sui-select-button:hover, .sui-select-button:focus {
  background: hsl(var(--background-step-3));
}

.sui-select-button.active,
.sui-select-button:active,
.sui-select-button:has(+ .sui-select-dropdown.active) {
  border: 1px solid hsl(var(--primary-base));
  background: hsl(var(--background-step-2));
}

.sui-select-dropdown {
  position: absolute;
  width: 100%;
  border: 1px solid hsl(var(--border));
  list-style: none;
  margin: 0;
  padding: 0;
  flex-direction: column;
  border-radius: var(--radius-md);
  background-color: hsl(var(--background-step-2));
  overflow: hidden;
  top: calc(100% + 0.25rem);
  left: 0;
  display: none;
  z-index: 90;
  box-shadow: 0px 4px 8px hsl(var(--shadow), 0.5);
}

.sui-select-dropdown.active {
  display: flex;
}

.sui-select-dropdown.above {
  top: auto;
  bottom: calc(100% - 18px + 0.25rem);
}

.sui-select-option {
  padding: 0.5rem;
  cursor: pointer;
  font-size: 0.975em;
  transition: all 0.15s ease;
}

.sui-select-option.disabled {
  pointer-events: none;
  color: hsl(var(--text-muted));
}

.sui-select-option:hover, .sui-select-option:focus, .sui-select-option.focused {
  background-color: hsl(var(--background-step-3));
}

.sui-select-option:focus {
  outline: none;
  border: none;
}

.sui-select-option.selected {
  background-color: hsl(var(--primary-base));
  color: hsl(var(--text-inverted));
  cursor: default;
}

.sui-hidden-select {
  height: 0;
  width: 0;
  border: none;
  outline: none;
  position: absolute;
  background-color: transparent;
  pointer-events: none;
  opacity: 0;
}
