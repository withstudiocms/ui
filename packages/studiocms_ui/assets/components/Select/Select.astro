---
import { generateID } from '../../utils/generateID.js';
import Icon from '../Icon/Icon.astro';
import './select.css';

/**
 * The props for the select component.
 */
interface Option {
	/**
	 * The label of the option.
	 */
	label: string;
	/**
	 * The value of the option.
	 */
	value: string;
	/**
	 * Whether the option is disabled. Defaults to `false`.
	 */
	disabled?: boolean;
}

/**
 * The props for the select component.
 */
interface Props {
	/**
	 * The label of the select.
	 */
	label?: string;
	/**
	 * The default value of the select.
	 */
	defaultValue?: string;
	/**
	 * Additional classes to apply to the select.
	 */
	class?: string;
	/**
	 * The name of the select. Required because of the helper.
	 */
	name?: string;
	/**
	 * Whether the select is required. Defaults to `false`.
	 */
	isRequired?: boolean;
	/**
	 * The options to display in the select.
	 */
	options: Option[];
	/**
	 * Whether the select is disabled. Defaults to `false`.
	 */
	disabled?: boolean;
	/**
	 * Whether the select is full width. Defaults to `false`.
	 */
	fullWidth?: boolean;
	/**
	 * The placeholder of the select. Defaults to `Select`.
	 */
	placeholder?: string;
}

const {
	label,
	defaultValue,
	class: className,
	name = generateID('select'),
	isRequired,
	options = [],
	disabled,
	fullWidth,
	placeholder = 'Select',
} = Astro.props;
---

<div
    id={`${name}-container`}
    class="sui-select-label"
    class:list={[disabled && "disabled", className, fullWidth && "full"]}
    data-options={JSON.stringify(options)}
    data-id={name}
>
    {label && (
        <label class="label" for={`${name}-select-btn`}>
            {label}
            <span class="req-star">{isRequired && "*"}</span>
        </label>
    )}
    <button
        class="sui-select-button"
        role="combobox"
        aria-controls={`${name}-dropdown`}
        aria-expanded="false"
        id={`${name}-select-btn`}
        type="button"
        aria-label={placeholder}
        title={placeholder}
    >
        <span class="sui-select-value-span" id={`${name}-value-span`}>
            {
                defaultValue
                    ? options.find((x) => x.value === defaultValue)?.label
                    : placeholder
            }
        </span>
        <Icon name="chevron-up-down" width={24} height={24} />
    </button>
    <ul class="sui-select-dropdown" role="listbox" id={`${name}-dropdown`}>
        {
            options.map((x, i) => (
                <li
                    class="sui-select-option"
                    role="option"
                    value={x.value}
                    class:list={[
                        defaultValue === x.value && `selected`,
                        x.disabled && "disabled",
                    ]}
                    id={defaultValue === x.value ? `${name}-selected` : ""}
                    data-option-index={i}
                >
                    {x.label}
                </li>
            ))
        }
    </ul>
    <select class="sui-hidden-select" id={name} name={name} required={isRequired} hidden tabindex="-1">
        <option value={""}> Select </option>
        {
            options.map((x) => (
                <option
                    value={x.value}
                    selected={defaultValue === x.value}
                    disabled={x.disabled}
                >
                    {x.label}
                </option>
            ))
        }
    </select>
</div>
<script>
import "studiocms:ui/scripts/select"
</script>
