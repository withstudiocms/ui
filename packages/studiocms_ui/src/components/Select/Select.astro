---
import { generateID } from '../../utils/generateID.js';
import Badge from '../Badge/Badge.astro';
import Icon from '../Icon/Icon.astro';
import './select.css';

/**
 * Represents an option in the select component.
 */
export interface SelectOption {
	/**
	 * The label of the option displayed to the user.
	 */
	label: string;
	/**
	 * The value of the option used for form submission.
	 */
	value: string;
	/**
	 * Whether the option is disabled. Defaults to `false`.
	 */
	disabled?: boolean;
}

/**
 * Base properties shared by both single and multiple select components.
 */
export interface SelectBaseProps {
	/**
	 * The label of the select field.
	 */
	label?: string;
	/**
	 * Additional classes to apply to the select component.
	 */
	class?: string;
	/**
	 * The name of the select field, used for form submission.
	 */
	name?: string;
	/**
	 * Whether the select field is required. Defaults to `false`.
	 */
	isRequired?: boolean;
	/**
	 * The options to display in the select dropdown.
	 */
	options: SelectOption[];
	/**
	 * Whether the select field is disabled. Defaults to `false`.
	 */
	disabled?: boolean;
	/**
	 * Whether the select field should take up the full width of its container. Defaults to `false`.
	 */
	fullWidth?: boolean;
	/**
	 * The placeholder text to display when no option is selected. Defaults to `Select`.
	 */
	placeholder?: string;
}

/**
 * Properties specific to single-select mode.
 */
export interface SingleSelectProps extends SelectBaseProps {
	/**
	 * Whether the select accepts multiple options.
	 */
	multiple?: false;
	/**
	 * The default selected value.
	 */
	defaultValue?: string;
	/**
	 * The maximum number of options that can be selected.
	 * This property has no effect in single-select mode.
	 */
	max?: never;
}

/**
 * Properties specific to multi-select mode.
 */
export interface MultiSelectProps extends SelectBaseProps {
	/**
	 * Whether the select accepts multiple options.
	 */
	multiple: true;
	/**
	 * The default selected values.
	 */
	defaultValue?: string[];
	/**
	 * The maximum number of options that can be selected.
	 */
	max?: number;
}

/**
 * Union type representing all possible select component properties.
 */
export type Props = SingleSelectProps | MultiSelectProps;

const {
	label,
	defaultValue,
	class: className,
	name = generateID('select'),
	isRequired,
	options = [],
	disabled,
	fullWidth,
	placeholder = 'Select',
	multiple = false,
	max = undefined,
} = Astro.props;

let selected: SelectOption | (SelectOption | undefined)[] | undefined;

if (multiple && Array.isArray(defaultValue)) {
	selected = defaultValue.map((x) => options.find((y) => y.value === x));
} else {
	selected = options.find((x) => x.value === defaultValue);
}

const defaultLabel = selected
	? Array.isArray(selected)
		? selected.map((x) => x?.label)
		: selected.label
	: placeholder;
---

<div
	id={`${name}-container`}
	class="sui-select-label"
	class:list={[disabled && "disabled", className, fullWidth && "full"]}
	data-options={JSON.stringify(options)}
	data-multiple={multiple ? "true" : undefined}
	data-multiple-max={multiple && max !== undefined ? max : undefined}
	data-id={name}
>
	{label && (
		<label class="label" for={`${name}-select-btn`}>
			{label}
			<span class="req-star">{isRequired && "*"}</span>
		</label>
	)}
	<div class="sui-select-dropdown-container">
		<button
			class="sui-select-button"
			role="combobox"
			aria-controls={`${name}-dropdown`}
			aria-expanded="false"
			id={`${name}-select-btn`}
			tabindex={disabled ? -1 : 0}
			type="button"
			aria-label={placeholder}
			title={placeholder}
		>
			<span class="sui-select-value-span" id={`${name}-value-span`}>
				{Array.isArray(selected)
					? <div class="sui-select-badge-container">{selected.map((s) => s && <Badge class="sui-select-badge" data-value={s.value} size="sm" label={s.label} iconPosition="right" icon="x-mark" />)}</div>
					: selected?.label ?? defaultLabel
				}
			</span>
			<Icon name="chevron-up-down" width={24} height={24} class="sui-select-chevron" />
		</button>
		<ul class="sui-select-dropdown" role="listbox" id={`${name}-dropdown`}>
			{
				options.map((x, i) => {
					const isSelected = Array.isArray(selected)
						? selected.map((y) => y && y.value).includes(x.value)
						: selected?.value === x.value;
					return (
						<li
							class="sui-select-option"
							role="option"
							value={x.value}
							class:list={[
								isSelected && `selected`,
								x.disabled && "disabled",
							]}
							id={isSelected ? `${name}-selected` : ""}
							data-option-index={i}
						>
							{x.label}
						</li>
					)
				})
			}
		</ul>
	</div>
	<select class="sui-hidden-select" id={name} name={name} required={isRequired} multiple={multiple ? "" : undefined} hidden tabindex="-1">
		<option value={""}> Select </option>
		{
			options.map((x) => {
				const isSelected = Array.isArray(selected)
					? selected.map((y) => y && y.value).includes(x.value)
					: selected?.value === x.value;
					
				return (
					<option
						value={x.value}
						selected={isSelected}
						disabled={x.disabled}
					>
						{x.label}
					</option>
				)
			})
		}
	</select>
	{multiple && max !== undefined && (
		<span class="sui-select-max-span">
			<span class="sui-select-select-count">0</span> / {max} selected
		</span>
	)}
	{
		multiple && Array.isArray(selected ?? []) && (		
			<div class="sui-select-badge-container-below"></div>
		)
	}
</div>
<script>
import "studiocms:ui/scripts/select"
</script>
