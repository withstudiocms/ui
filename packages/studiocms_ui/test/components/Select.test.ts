import { describe, expect } from 'vitest';
import Select from '../../src/components/Select/Select.astro';
import { test } from '../fixtures/vitest/AstroContainer';

const mockProps = {
	name: 'select',
	label: 'Select Element',
	options: [
		{ label: 'Option 1', value: 'opt-1' },
		{ label: 'Option 2', value: 'opt-2' },
		{ label: 'Option 3', value: 'opt-3' },
	],
};

describe('Select Component', () => {
	test('renders Select component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Select, 'Select', { props: mockProps });
		expect(result).toMatchSnapshot();
	});
});
