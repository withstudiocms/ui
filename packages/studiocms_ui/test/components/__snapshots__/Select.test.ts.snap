// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select Component > renders Select component correctly 1`] = `"<div id="select-container" class="sui-select-label" data-options="[{&#34;label&#34;:&#34;Option 1&#34;,&#34;value&#34;:&#34;opt-1&#34;},{&#34;label&#34;:&#34;Option 2&#34;,&#34;value&#34;:&#34;opt-2&#34;},{&#34;label&#34;:&#34;Option 3&#34;,&#34;value&#34;:&#34;opt-3&#34;}]" data-id="select"> <label class="label" for="select-select-btn"> Select Element <span class="req-star"></span> </label> <div class="sui-select-dropdown-container"> <button class="sui-select-button" role="combobox" aria-controls="select-dropdown" aria-expanded="false" id="select-select-btn" tabindex="0" type="button" aria-label="Select" title="Select"> <span class="sui-select-value-span" id="select-value-span"> Select </span> <svg style="min-width: 24px" xmlns="http://www.w3.org/2000/svg" prefix="heroicons" class="sui-select-chevron" height="24" width="24" viewbox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M8.25 15L12 18.75L15.75 15m-7.5-6L12 5.25L15.75 9"/></svg> </button> <div class="sui-select-dropdown"> <ul class="sui-select-dropdown-list" role="listbox" id="select-dropdown"> <li role="option" value="opt-1" class="sui-select-option" id data-option-index="0"> Option 1 </li><li role="option" value="opt-2" class="sui-select-option" id data-option-index="1"> Option 2 </li><li role="option" value="opt-3" class="sui-select-option" id data-option-index="2"> Option 3 </li> </ul> </div> </div> <select class="sui-hidden-select" id="select" name="select" hidden tabindex="-1"> <option value> Select </option> <option value="opt-1"> Option 1 </option><option value="opt-2"> Option 2 </option><option value="opt-3"> Option 3 </option> </select>   </div> <script type="module" src="/mock/path/Select.astro?astro&type=script&index=0&lang.ts"></script>"`;
